test/browser/call-joining.js

```diff
@@ -0,0 +1,147 @@
+import config from './config.js'
+
+const assert = chai.assert;
+let client;
+let peerUser1, peerUser2;
+let call;
+describe('Call Joining', async function() {
+    this.timeout(300000);
+
+    before(async function() {
+        Circuit.logger.setLevel(Circuit.Enums.LogLevel.Error);
+        client = new Circuit.Client(config.config);
+        const res = await Promise.all([PeerUser.create(), PeerUser.create(), client.logon(config.credentials)]);
+        peerUser1 = res[0];
+        peerUser2 = res[1];
+        const conversation = await client.createGroupConversation([peerUser2.userId], 'SDK Test: Conference Call');
+        call = await client.startConference(conversation.convId, {audio: true, video: false});
+        await expectEvents(client, [{
+            type: 'callStatus',
+            predicate: evt => evt.call.state === Circuit.Enums.CallStateName.Initiated
+        }, {
+            type: 'callStatus',
+            predicate: evt => evt.call.state === Circuit.Enums.CallStateName.Waiting
+        }]);
+        await sleep(5000); // wait to make sure the call is ready to be joined
+    });
+
+    after(async function() {
+        await Promise.all([peerUser1.destroy(), peerUser2.destroy(), client.logout()]);
+    });
+
+    afterEach(async function() {
+        client.removeAllListeners();
+    });
+
+    it('should get started calls', async () => {
+        const res = await peerUser2.exec('getStartedCalls');
+        assert(res.some(c => c.callId === call.callId));
+    })
+
+    it('should add participant to call with addParticipantToCall and check they are a guest', async () => {
+        await client.addParticipantToCall(call.callId, { userId: peerUser1.userId });
+        await sleep(5000);
+        await Promise.all([
+            peerUser1.exec('answerCall', call.callId, {audio: false, video: false}),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'callStateChanged' && evt.call.state === Circuit.Enums.CallStateName.Active
+            }, {
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantJoined'
+            }])
+        ]);
+        await sleep(5000);
+        call = await client.findCall(call.callId);
+        assert(call.participants.some(user => user.userId === peerUser1.userId && user.isMeetingGuest));
+    });
+
+    it('should allow other user to join the conference', async () => {
+        await Promise.all([
+            peerUser2.exec('joinConference', call.callId, {audio: true, video: false}),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'callStateChanged' && evt.call.state === Circuit.Enums.CallStateName.Active
+            }, {
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantJoined'
+            }])
+        ]);
+        await sleep(5000);
```

Comment:
I don't think we need this. We already verify that the event with participantJoined is received. We can add the "call.participants.some..." check in the predicate.

No need to wait and find the call.